src/include/txn.i

```diff
@@ -851,24 +851,20 @@ __wt_txn_read_upd_list(
         if (type == WT_UPDATE_RESERVE)
             continue;
 
+        if (type == WT_UPDATE_TOMBSTONE && F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE) &&
+          !__wt_txn_upd_visible_all(session, upd)) {
```

Comment:
Why don't do this same thing when we see a globally visible tombstone? Do we then know that `__wt_txn_upd_visible_type` will return `WT_VISIBLE_TRUE`, and the caller will handle the tombstone in a meaningful way?

Reply:
I had quick look how callers for `__wt_txn_upd_visible_type()` have been handling this and I can confirm you are correct. All callers are managing the scenario in some special manner when the call returns a tombstone update.

Reply:
understood.